File: src/main/webui/src/model/common/ComputationResultFree.ts
import ModifiedValue from "./ModifiedValue";
import Research from "./Research";

export default class ComputationResult {
    finalProperties: ModifiedValue[];
    requiredResearch: Research[];
    customizationPoint: number;
    customisationBasePoint: number;
    constructor(finalProperties: ModifiedValue[], requiredResearch: Research[], customizationPoint: number, customisationBasePoint: number) {
        this.finalProperties = finalProperties;
        this.requiredResearch = requiredResearch;
        this.customizationPoint = customizationPoint;
        this.customisationBasePoint = customisationBasePoint;
    }
}

File: src/main/java/io/tbbc/cf/turret/computation/ComputationResultFree.java
package io.tbbc.cf.turret.computation;

import io.tbbc.cf.property.FinalProperties;
import io.tbbc.cf.research.Research;

import java.util.List;

public record ComputationResultFree(FinalProperties finalProperties, List<Research> requiredResearch,
                                    int customizationPoint, int customisationBasePoint) {
}
